spec/searchBarSubMenu.component.spec.js

```diff
@@ -0,0 +1,112 @@
+let viewName = "TEST";
+let searchBarSubMenuItems = [
+  {
+    name: "Back to Primo Classic",
+    description: "Back to Primo Classic",
+    action: "http://primo.school.edu/primo_library/libweb/action/search.do?vid=" + viewName,
+    icon: {
+      set: 'navigation',
+      icon: 'ic_arrow_back_24px'
+    },
+    show_xs: true,
+    cssClasses: 'button-over-dark'
+  },
+  {
+    name: "Library Hours",
+    description: "Library Hours",
+    action: "https://school.edu/library-hours",
+    icon: {
+      set: 'av',
+      icon: 'ic_av_timer_24px'
+    },
+    cssClasses: 'button-over-dark'
+  }
+];
+
+describe('searchBarSubMenu component', () => {
+  beforeEach(module('searchBarSubMenu', ($provide) => {
+    $provide.constant("searchBarSubMenuItems", searchBarSubMenuItems);
+    $provide.value("translateFilter", (original) => original + "!");
+  }));
+
+  let $compile, $rootScope, element;
+  beforeEach(inject(function(_$compile_, _$rootScope_){
+    $compile = _$compile_;
+    $rootScope = _$rootScope_;
+    element = $compile("<search-bar-sub-menu></search-bar-sub-menu>")($rootScope);
+    // element is a jqlite object
+    $rootScope.$digest();
+  }));
+
+  describe('template layout', () => {
+
+    it('should have styled div element at the top-level', () => {
+      const children = element.children();
+      const div = children[0];
+
+      expect(div.tagName).toEqual('DIV');
+      expect(children.length).toEqual(1);
+      expect(div.className).toContain('search-bar-sub-menu');
+    });
+
+    it('should contain a single unordered list', () => {
+      const uls = element.find('ul');
+      expect(uls.length).toEqual(1);
+    });
+  });
+
+  describe('submenu buttons', () => {
+    let buttons;
+    beforeEach(() => {
+      buttons = element.find('ul')[0].querySelectorAll('button');
+    });
+
+    it('should be appropriate amount', () => {
+      expect(buttons.length).toEqual(searchBarSubMenuItems.length);
+    });
+
+    it('should include name and description', () => {
+      for(let i = 0; i < buttons.length; i++) {
+        const button = buttons[i];
+        expect(button.innerText).toContain(searchBarSubMenuItems[i].name);
+        expect(button.innerText).toContain(searchBarSubMenuItems[i].description);
+      }
+    });
+
+    it('should link to action property', () => {
+      for(let i = 0; i < buttons.length; i++) {
```

Comment:
@etgrieco why `for` over `array.forEach`? is it just a performance thing? i'm not partial to either one.

Reply:
I prefer forEach, but buttons is a NodeList object and both Array.from and NodeList.forEach methods weren't being polyfilled with es-2015 preset